src/Database/Builder.php

```diff
@@ -153,7 +154,7 @@ public function paginate($perPage = null, $currentPage = null, $columns = ['*'],
             $perPage = $this->model->getPerPage();
         }
 
-        $total = $this->toBase()->getCountForPagination();
+        $total = value($total) ?? $this->toBase()->getCountForPagination();
```

Comment:
I don't think I've seen this function before, what does it do and is it better than just doing `$total ?? $this->toBase()->getCountForPagination();`?

Reply:
I took this from Laravel `Builder::paginate()`.